src/controllers/auth-controller.ts

```diff
@@ -0,0 +1,95 @@
+import User from "../../src/model/user-model";
+import catchAsync from "../../utils/catch-async";
+import { Request, Response, NextFunction } from "express";
+import { customError } from "../../errors/error";
+
+//////
+const jwt = require("jsonwebtoken");
+
+////
+export const signUp = catchAsync(
+  async (req: Request, res: Response, next: NextFunction) => {
+    const {
+      email,
+      password,
+      name,
+    }: { email: string; password: string; name: string } = req.body;
+
+    const user = await User.create({
+      email,
+      password,
+      name,
+    });
+    const token = jwt.sign({ id: user._id }, process.env.JWT_KEY, {
+      // expiresIn: process.env.JWT_EXPIRES_IN,
```

Comment:
You should probably be using the EXPIRES for JWT, is this meant to be commented out

Reply:
It was a mistake, i have fixed that too